spark/src/main/scala/geotrellis/spark/knn/KNearestRDD.scala

```diff
@@ -0,0 +1,27 @@
+package geotrellis.spark.knn
+
+import geotrellis.spark._
+import geotrellis.vector._
+import com.vividsolutions.jts.geom.Envelope
+import com.vividsolutions.jts.geom.Coordinate
+
+import org.apache.spark.rdd.RDD
+
+object KNearestRDD {
+  def kNearest[T](rdd: RDD[T], x: Double, y: Double, k: Int)(f: T => Extent): List[T] =
+    kNearest(rdd, new Envelope(new Coordinate(x, y)), k)(f)
```

Comment:
We should not use naked `jts` types, that way madness lies. Results of most operations is a `Geometry`, which then requires you to test for types through reflection. This is one of the main reasons we ended up wrapping it, so we can use scala type system to reason about the result types: https://github.com/geotrellis/geotrellis/blob/master/vector/src/main/scala/geotrellis/vector/Results.scala.